### PR TITLE
Migrate testRunner from jasmine2 to jest-circus

### PR DESCRIPTION
**Pull request by ymqy**
## Summary

In jest v27, jest-circus as default test runner (https://github.com/facebook/jest/pull/10686)

## How did you test this change?

ci green

### CODE REVIEW

File: scripts/jest/spec-equivalence-reporter/equivalenceReporter.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+'use strict';
+
+class EquivalenceReporter {
+  onTestCaseResult(test, testCaseResult) {
+    console.log(
+      `EQUIVALENCE: ${testCaseResult.title}, ` +
+        `status: ${testCaseResult.status}, ` +
+        `numExpectations: ${testCaseResult.numPassingAsserts}`
```
Comment (ymqy) [inline scripts/jest/spec-equivalence-reporter/equivalenceReporter.js:15]: In Jest 29, the testCaseResult.numPassingAsserts always returns 0. I submitted a PR https://github.com/facebook/jest/pull/13795 to implement the return of the actual value

File: packages/react-client/src/__tests__/ReactFlight-test.js [hunk 2]
```diff
@@ -240,7 +244,7 @@ describe('ReactFlight', () => {
       ReactNoop.render(rootModel);
     });
     expect(ReactNoop).toMatchRenderedOutput('Loading...');
-    spyOnDevAndProd(console, 'error');
+    spyOnDevAndProd(console, 'error').mockImplementation(() => {});
```
Comment (ymqy) [inline packages/react-client/src/__tests__/ReactFlight-test.js:247]: This approach may need further improvement

Comment (ymqy) [reply packages/react-client/src/__tests__/ReactFlight-test.js:247]: There is a significant difference between Jasmine's spyOn and jest.spyOn, jest.spyOn not only spies on the method but also calls the spied method, whereas Jasmine's spyOn does not call the spied method.

Comment (eps1lon) [reply packages/react-client/src/__tests__/ReactFlight-test.js:247]: Makes sense. So this change is just ensuring equivalence between using jasmine2 and jest-circus, right?

Comment (ymqy) [reply packages/react-client/src/__tests__/ReactFlight-test.js:247]: Yes, that's correct. However, I feel that this implementation is not elegant.

**Review by eps1lon — APPROVED** [review]

**Comment by kassens** [comment]
Awesome!

File: scripts/jest/spec-equivalence-reporter/equivalenceReporter.js (hunk 1 above)
Comment (kassens) [reply scripts/jest/spec-equivalence-reporter/equivalenceReporter.js:15]: Thanks for the fix, pulling that update into React with https://github.com/facebook/react/pull/26150

Comment (ymqy) [reply scripts/jest/spec-equivalence-reporter/equivalenceReporter.js:15]: The fix for numPassingAsserts has been released in v29.4.1 😂

**Pull request merged**